CONTRIBUTING.md

```diff
@@ -198,22 +198,29 @@ Add them for all new code, with the exception of:
 Make sure you import the types from `net.jcip`, there are homonyms in the classpath.
 
 
+### Nullability annotations
+
+We use the [Spotbugs annotations](https://spotbugs.github.io) to document nullability of parameters,
+method return types and class members.
+
+Please annotate any new class or interface with the appropriate annotations. Make sure you import 
+the types from `edu.umd.cs.findbugs.annotations`, there are homonyms in the classpath.
+
+
 ## Coding style -- test code
 
 Static imports are permitted in a couple of places:
-* AssertJ's `assertThat` / `fail`.
-* Some Mockito methods, provided that you're already using a non-statically imported method at the
-  beginning of the line. For example:
+* AssertJ methods whose names start with `assertThat` or `fail`.
+* Mockito methods, e.g.:
   ```java
-  // any and eq are statically imported, it's pretty clear that they at least relate to Mockito
-  Mockito.verify(intCodec).decodePrimitive(any(ByteBuffer.class), eq(ProtocolVersion.DEFAULT));
+  verify(intCodec).decodePrimitive(any(ByteBuffer.class), eq(ProtocolVersion.DEFAULT));
```

Comment:
Sounds good to me. Again Mockito is pretty easy to recognize from the context once you're familiar with it, and we can take a bit more liberty in test code.

But then for consistency we should do a pass on existing code and static import it everywhere. I'm not sure if there's a smart way to do that in IDEA.

Reply:
> But then for consistency we should do a pass on existing code and static import it everywhere. I'm not sure if there's a smart way to do that in IDEA.

Yes I was planning to do this in a separate commit once we agree on the guidelines. It will probably be a mix of automatic search-replace and manual editing.